pkg/services/baremetal/host/host.go

```diff
@@ -91,6 +91,29 @@ func NewService(s *scope.BareMetalHostScope) *Service {
 
 // Reconcile implements reconcilement of HetznerBareMetalHosts.
 func (s *Service) Reconcile(ctx context.Context) (result reconcile.Result, err error) {
+	// Fetch the Hetzner bare metal host instance.
+	bmHost := &infrav1.HetznerBareMetalHost{}
+	err = s.scope.Client.Get(ctx, client.ObjectKey{
+		Namespace: s.scope.HetznerBareMetalHost.Namespace,
+		Name:      s.scope.HetznerBareMetalHost.Name,
+	}, bmHost)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get HetznerBareMetalHost : %w", err)
+	}
+	if !bmHost.DeletionTimestamp.IsZero() {
```

Comment:
you need to put this code after the "oldHost" is defined. The oldHost is the "original" state of the host and all changes thereafter are updated via the "Update" API call below